js/dragonex.js

```diff
@@ -0,0 +1,912 @@
+        const markets = this.safeValue (spotMarkets, 'list', []);
+        for (let i = 0; i < markets.length; i++) {
+            const market = markets[i];
+            const id = market[1];
+            const parts = id.split ('_');
+            const baseId = parts[0];
+            const quoteId = parts[1];
+            const base = this.safeCurrencyCode (baseId);
+            const quote = this.safeCurrencyCode (quoteId);
+            const precision = {
+                'amount': market[7],
+                'price': market[5],
+            };
+            const symbol = base + '/' + quote;
+            const taker = this.fees['trading']['taker'];
+            result.push ({
+                'id': id,
+                'symbol': symbol,
+                'symbol_id': market[0],
+                'base': base,
+                'quote': quote,
+                'baseId': baseId,
+                'quoteId': quoteId,
+                'active': true,
+                'precision': precision,
+                'taker': taker,
+                'limits': {
+                    'amount': {
+                        'min': market[4],
+                        'max': undefined,
+                    },
+                    'price': {
+                        'min': market[2],
+                        'max': undefined,
+                    },
+                    'cost': {
+                        'min': parseFloat (market[4]) * parseFloat (market[2]),
+                        'max': undefined,
+                    },
+                },
+            });
+        }
+        return result;
+    }
+
+    calculateFee (symbol, type, side, amount, price, takerOrMaker = 'taker', params = {}) {
+        const market = this.markets[symbol];
+        let key = 'quote';
+        const rate = market[takerOrMaker];
+        let cost = amount * rate;
+        let precision = market['precision']['price'];
+        if (side === 'sell') {
+            cost *= price;
+        } else {
+            key = 'base';
+            precision = market['precision']['amount'];
+        }
+        cost = this.decimalToPrecision (cost, ROUND, precision, this.precisionMode);
+        return {
+            'type': takerOrMaker,
+            'currency': market[key],
+            'rate': rate,
+            'cost': parseFloat (cost),
+        };
+    }
+
+    async fetchBalance (params = {}) {
+        await this.loadMarkets ();
+        const response = await this.privatePostUserOwn (params);
+        const result = { 'info': response };
+        const balances = this.safeValue (response, 'data', []);
+        for (let i = 0; i < balances.length; i++) {
+            const balance = balances[i];
+            const currencyId = balance['coin_id'];
+            const code = this.safeCurrencyCode (currencyId);
+            const account = this.account ();
+            account['free'] = this.safeFloat (balance, 'volume') - this.safeFloat (balance, 'frozen');
```

Comment:
This calculation is not necessary, because `free` will be automatically calculated from `used` and `total` if you simply omit it. See how this is implemented with other exchanges.